### PR TITLE
feat: add nano contract class with serializer/deserializer methods

### PR DESCRIPTION
**Pull request by pedroferreira1**
## Acceptance Criteria

- Create NanoContract class.
- Add methods to serialize and deserialize the nano contract object.
- Transaction mining service should be able to mine nano contract transactions with the methods in the class.

### CODE REVIEW

File: hathorlib/nanocontracts/nanocontract.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    """This tuple carries the pieces of information serialized inside transactions."""
+    version: int
+    id: bytes
+    method: str
+    args_bytes: bytes
+    pubkey: bytes
+    signature: bytes
+
+
+class NanoContract(Transaction):
```
Comment (msbrogli) [inline hathorlib/nanocontracts/nanocontract.py:36]: I miss unit tests for the serialization and deserialization methods.

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: hathorlib/nanocontracts/nanocontract.py (hunk 1 above)
Comment (pedroferreira1) [reply hathorlib/nanocontracts/nanocontract.py:36]: I will add them and fix the linter

**Review by msbrogli — COMMENTED** [review]
This is the first EVER public PR related to Nano Contracts! Great job, @pedroferreira1 ! \o/ \o/ \o/

File: hathorlib/nanocontracts/nanocontract.py (hunk 1 above)
Comment (pedroferreira1) [reply hathorlib/nanocontracts/nanocontract.py:34]: Done

Comment (msbrogli) [reply hathorlib/nanocontracts/nanocontract.py:34]: Thanks!

**Pull request merged**